### PR TITLE
Use one full-window running baseline for BCO gradient accumulation

### LINKED ISSUE #7265: [Bug] BCO uses different running baselines within one gradient-accumulation window
### Reproduction

`BCOTrainer` updates `RunningMoments` and reads its mean inside every `bco_loss` call:

```python
if do_train:
    self.running.update(torch.cat((chosen_rewards, rejected_rewards), 0).detach())
delta = torch.as_tensor(self.running.mean, device=chosen_rewards.device)
```

With gradient accumulation, the microbatches of one optimizer step therefore use different baselines. The final running mean is the same once all samples have been seen; the problem is the sequence of baselines used for the gradients along the way.

To reproduce, use a deterministic small GPT-2 with dropout off, the same 16 ordered examples (chosen/rejected order fixed), the same policy and reference checkpoints, and `RunningMoments` initialized to count 16, mean `0.2`, variance `0.25`. Do one optimizer update at learning rate `1e-6` through `trl.experimental.bco.BCOTrainer.train`, and record the baseline used by each `bco_loss` call:

- one batch with `gradient_accumulation_steps=1`;
- four microbatches with `gradient_accumulation_steps=4`.

| One batch vs. four accumulated microbatches | Difference |
| --- | ---: |
| Full gradient, relative L2 | `0.03228330` |
| Parameter update, relative L2 | `0.06736848` |
| Parameter update, max abs | `1.9967556e-6` |

Updating `RunningMoments` once from all rewards in the window and reusing the resulting mean for every microbatch gives identical parameter updates. A fixed-baseline BCO run and a DPO run agree across the two layouts. Reproduced on a second host.

Expected behavior: the accumulated run should use the same baseline as the unsplit run, which is the mean **after** the whole window's rewards are incorporated (not the previous window's mean, which would change the current unsplit semantics).

The version I tested does a no-grad reward pass over the pending microbatches in `get_batch_samples`, commits `RunningMoments` once, and then computes the normal microbatch losses without further updates. It adds forward work, and a production version needs to handle dropout/RNG restoration, partial windows and multi-rank reward aggregation. The numbers above are a deterministic one-update FP32 test, not a preference-quality result.

### System Info

```text
TRL: 1.12.0 at 59c4a8e104413fa9f4ca1a54eaf2ff93c0f299be
Python: 3.11.15
PyTorch: 2.13.0+cu130
Transformers: 5.15.0
Accelerate: 1.14.0
Platform: Linux, CUDA GPU
```

Source: [bco_trainer.py at 59c4a8e1](https://github.com/huggingface/trl/blob/59c4a8e104413fa9f4ca1a54eaf2ff93c0f299be/trl/experimental/bco/bco_trainer.py).

### Checklist

- [x] I have checked that my issue isn't already filed
- [x] I have included my system information
- [ ] Any code provided is minimal, complete, and reproducible
- [x] Any code provided is properly formatted in code blocks
- [x] Any traceback provided is complete (no traceback; this is a silent numerical issue)

### PR DESCRIPTION
**Pull request by gss10282025**
# What does this PR do?

Collect detached rewards for the whole optimizer window, update `RunningMoments` once, and let every microbatch read the same updated mean. This keeps the unsplit BCO objective while allowing gradient accumulation.

The pending microbatches are collected in `get_batch_samples`, their rewards are computed without gradients, and running-statistic writes are suppressed during the loss calls. Evaluation stays read-only. Freezing the old mean and committing after backward would be a different objective, so that is not what this does.

At TRL 1.12.0 the one-vs-four-microbatch parameter-update difference drops from `0.06736848` relative L2 to zero; fixed-baseline and DPO controls agree across layouts; reproduced on a second host. The collection pass restores RNG state and registered model buffers. Full Trainer/GPU execution and multi-rank aggregation still need validation.

## Before submitting

- [ ] This PR fixes a typo or improves the docs
- [x] Did you read the contributor guideline, Pull Request section?
- [x] Was this discussed/approved via a GitHub issue? Reported in #7265; no maintainer approval is implied.
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

## AI writing disclosure

- [ ] No AI usage: the PR was written entirely by a human.
- [ ] AI-assisted: some parts were suggested or improved by AI, but the PR was written and reviewed by a human.
- [x] AI-generated: the PR was mostly or fully generated by an AI tool.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed.

## Checks on this branch

Added `tests/experimental/test_bco_window.py`. Its isolated CPU execution passed using the production prefetch/loss methods and RunningMoments: full, equal and uneven windows agree in baseline and gradient; the prepass restores buffers and RNG; evaluation does not update the running state. This isolates the window logic and does not run a complete Trainer job. Ruff, Python syntax and `git diff --check` pass.

Fixes #7265. The separate removal proposal #7136 is still open; this draft addresses the behavior while BCO remains available.

### CODE REVIEW

**Comment by qgallouedec** [comment]
Closing in favour of #7136, which removes `trl.experimental.bco`.
